File: app/src/core/tuning/cephPgCalculator.test.ts
import { describe, expect, it } from 'vitest'
import { CephPgCalculator } from './cephPgCalculator'

describe('CephPgCalculator', () => {
  const calculator = new CephPgCalculator()

  it('rounds up to the nearest power of 2', () => {
    const result = calculator.execute({
      osdCount: 20,
      poolCount: 1,
      targetPgsPerOsd: 100,
      replicationFactor: 3,
    })

    expect(result.rawPgCount).toBeCloseTo(666.67, 2)
    expect(result.roundedPgCount).toBe(1024)
  })

  it('rejects non-positive inputs', () => {
    expect(() =>
      calculator.execute({
        osdCount: 0,
        poolCount: 1,
        targetPgsPerOsd: 100,
        replicationFactor: 3,
      }),
    ).toThrow()
  })
})

File: archive/flutter-app/test/core/tuning/ceph_pg_calculator_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:infrakit_studio/core/tuning/ceph_pg_calculator.dart';

void main() {
  const calculator = CephPgCalculator();

  test('rounds up to the nearest power of 2', () {
    final result = calculator.execute(
      const CephPgInput(
        osdCount: 20,
        poolCount: 1,
        targetPgsPerOsd: 100,
        replicationFactor: 3,
      ),
    );

    expect(result.rawPgCount, closeTo(666.67, 0.01));
    expect(result.roundedPgCount, 1024);
  });

  test('rejects non-positive inputs', () {
    expect(
      () => calculator.execute(
        const CephPgInput(
          osdCount: 0,
          poolCount: 1,
          targetPgsPerOsd: 100,
          replicationFactor: 3,
        ),
      ),
      throwsArgumentError,
    );
  });
}
